x/concentrated-liquidity/msg_server.go

```diff
@@ -118,9 +118,13 @@ func (server msgServer) CollectFees(goCtx context.Context, msg *types.MsgCollect
 		return nil, err
 	}
 
-	collectedFees, err := server.keeper.collectFees(ctx, sender, msg.PositionId)
-	if err != nil {
-		return nil, err
+	totalCollectedFees := sdk.NewCoins()
+	for _, positionId := range msg.PositionIds {
+		collectedFees, err := server.keeper.collectFees(ctx, sender, positionId)
+		if err != nil {
+			return nil, err
+		}
+		totalCollectedFees = totalCollectedFees.Add(collectedFees...)
```

Comment:
It might make sense to emit events to reflect how much was emitted by each position id in addition to the total.

Might help with debugging if something horribly wrong ever occurs.

Reply:
Good idea! I added this here, do you mind reviewing this code before I merge? https://github.com/osmosis-labs/osmosis/pull/4762/commits/bf75084518dc1ed3a1485670825356d8e444a931

Reply:
Similar to: https://github.com/osmosis-labs/osmosis/pull/4762#discussion_r1151172339 but this is not blocking